src/core/Akka.Remote.Tests.MultiNode/RemoteNodeRestartGateSpec.cs

```diff
@@ -0,0 +1,178 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="RemoteNodeRestartGateSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2016 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2016 Akka.NET project <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Text;
+using System.Text.RegularExpressions;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.Remote.TestKit;
+using Akka.Remote.Transport;
+using Akka.TestKit;
+using Akka.Util.Internal;
+
+namespace Akka.Remote.Tests.MultiNode
+{
+
+    #region Spec
+
+    public abstract class RemoteNodeRestartGateSpec : MultiNodeSpec
+    {
+        private readonly RemoteNodeRestartGateSpecConfig _specConfig;
+
+        protected RemoteNodeRestartGateSpec()
+            : this(new RemoteNodeRestartGateSpecConfig())
+        {
+        }
+
+        protected RemoteNodeRestartGateSpec(RemoteNodeRestartGateSpecConfig specConfig)
+            : base(specConfig)
+        {
+            _specConfig = specConfig;
+        }
+
+        protected override int InitialParticipantsValueFactory
+        {
+            get { return 2; }
+        }
+
+        private IActorRef Identify(RoleName role, string actorName)
+        {
+            Sys.ActorSelection(Node(role)/"user"/actorName).Tell(new Identify(actorName));
+            return ExpectMsg<ActorIdentity>().Subject;
+        }
+
+
+        [MultiNodeFact]
+        public void RemoteNodeRestart_must_allow_restarted_node_to_pass_through_gate()
+        {
+            Sys.ActorOf(Props.Create(() => new Subject()), "subject");
+            EnterBarrier("subject-started");
+
+            RunOn(() =>
+            {
+                var secondAddress = Node(_specConfig.Second).Address;
+
+                Identify(_specConfig.Second, "subject");
+
+                EventFilter.Warning(new Regex("address is now gated")).ExpectOne(() =>
+                {
+                    RARP.For(Sys).Provider.Transport.ManagementCommand(
+                        new ForceDisassociateExplicitly(Node(_specConfig.Second).Address, DisassociateInfo.Unknown))
+                        .Wait(TimeSpan.FromSeconds(3));
+                });
+
+
+                EnterBarrier("gated");
+                TestConductor.Shutdown(_specConfig.Second).Wait();
+                Within(TimeSpan.FromSeconds(10), () =>
+                    {
+                        AwaitAssert(
+                            () =>
+                                {
+                                    Sys.ActorSelection(new RootActorPath(secondAddress) / "user" / "subject")
+                                        .Tell(new Identify("subject"));
+                                    ExpectMsg<ActorIdentity>().Subject.ShouldNotBe(null);
```

Comment:
This line needed to be changed to perform an assertion on `ActorIdentity.Subject` to make sure that it was non-null, otherwise node 1 would bail out early.